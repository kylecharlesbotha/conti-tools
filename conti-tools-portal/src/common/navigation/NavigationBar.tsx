import MenuIcon from '@mui/icons-material/Menu';
import { AppBar, IconButton, Theme, Toolbar, Typography } from '@mui/material';
interface NavigationBarProps {
  toggleIsOpen: () => void;
  pageTitle: string;
  isOpen: boolean;
  drawerWidth: number;
}

export const NavigationBar = ({
  toggleIsOpen,
  pageTitle,
  isOpen,
  drawerWidth
}: NavigationBarProps) => {
  return (
    <AppBar
      position="fixed"
      sx={{
        zIndex: (theme) => theme.zIndex.drawer + 1,
        transition: (theme: Theme) =>
          theme.transitions.create(['width', 'margin'], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.leavingScreen
          }),
        ...(isOpen && {
          marginLeft: drawerWidth,
          width: `calc(100% - ${drawerWidth}px)`,
          transition: (theme: Theme) =>
            theme.transitions.create(['width', 'margin'], {
              easing: theme.transitions.easing.sharp,
              duration: theme.transitions.duration.enteringScreen
            })
        })
      }}
    >
      <Toolbar>
        <IconButton
          size="large"
          edge="start"
          color="inherit"
          aria-label="menu"
          sx={{ mr: 2 }}
          onClick={() => toggleIsOpen()}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
          {pageTitle}
        </Typography>
      </Toolbar>
    </AppBar>
  );
};
