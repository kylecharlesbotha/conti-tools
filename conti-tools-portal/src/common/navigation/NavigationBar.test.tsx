import { fireEvent, render, screen } from '@testing-library/react';
import { NavigationBar } from './NavigationBar';

describe('Testing navigation bar', () => {
  test('should render correctly', () => {
    const { baseElement } = render(
      <NavigationBar
        isOpen={true}
        drawerWidth={240}
        toggleIsOpen={jest.fn}
        pageTitle={'Home'}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
  test('should call the toggle function when the button is clicked', () => {
    const mockToggle = jest.fn();
    render(
      <NavigationBar
        isOpen={true}
        drawerWidth={240}
        toggleIsOpen={mockToggle}
        pageTitle={'Home'}
      />
    );
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockToggle).toHaveBeenCalled();
  });
});
