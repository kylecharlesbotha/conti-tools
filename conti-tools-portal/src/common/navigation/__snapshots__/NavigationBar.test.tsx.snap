// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing navigation bar should render correctly 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1sfranv-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          Home
        </div>
      </div>
    </header>
  </div>
</body>
`;
